### PR TITLE
Fix iceberg full refresh#247

### LINKED ISSUE #247: dbt --full-refresh not working with iceberg incremental model
### Describe the bug
The dbt command flag **--full-refresh** is not working with iceberg incremental model

### Steps To Reproduce
Create a first model with the following config
```yml
{{
    config(
        materialized='incremental',
        incremental_strategy='merge',
        unique_key=['first_name', 'last_name'],
        file_format= 'iceberg'
    )
}}
```
Output:
Capture d’écran 2023-10-24 à 17 09 44

Now **we add a new column to the source table data**, and we run our target table in full refresh mode. We make sure to select the new column.

`dbt -d run -f -t dev_glue -s glue_incremental_merge`

The glue_incremental_merge table does not get updated with the new column.
We get the same output as before:

Capture d’écran 2023-10-24 à 17 32 25

In the detailed logs, we can see that, despite the -f flag, a merge was performed.

Capture d’écran 2023-10-24 à 17 29 12

### Expected behavior
As per [dbt-core documentation](https://docs.getdbt.com/docs/build/incremental-models#how-do-i-rebuild-an-incremental-model):
To force dbt to rebuild the entire incremental model from scratch, use the --full-refresh flag on the command line. This flag will cause dbt to drop the existing target table in the database before rebuilding it for all-time.

### System information
dbt-glue: 1.6.5
python: 3.9.11

### PR DESCRIPTION
**Pull request by sanga8**
Resolves #247

### Description

Fixing [full refresh with iceberg issue](https://github.com/aws-samples/dbt-glue/issues/247).

In the `incremental.sql` file, it was first checking if the relation was iceberg, instead of checking for full refresh.
See [athena connector implementation](https://github.com/dbt-athena/dbt-athena/blob/main/dbt/include/athena/macros/materializations/models/incremental/incremental.sql) for reference.

We fix it by modifying the order of the condition, we first check if the user has used the full refresh flag (i.e. `-f` or `--full-refresh`).

To check if the user used the flag, we use the dbt core macro [should_full_refresh()](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/configs.sql#L6) instead of the currently implemented *full_refresh_mode* variable. Since we don't need the *full_refresh_mode* variable anymore, we remove this variable from the code.

Once the full refresh condition is True, we check if it is an iceberg materialization. If yes, we first delete the table using `glue__drop_relation` macro in adapters.sql. Then, we write the table with the `iceberg_write` function, that will re-create the table from scratch since it does not exist anymore.

TLDR:
- We modify the order of condition by first checking if full refresh is enabled instead of if iceberg.
- We check if full refresh is enabled with a dbt-core macro instead of custom implementation.
- If full refresh and iceberg, then we delete the table.
- We re-write the table using iceberg write.

### Checklist

- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-glue next" section.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dbt/include/glue/macros/materializations/incremental/incremental.sql [hunk 1]
```diff
@@ -57,17 +47,29 @@
...
+        {% elif file_format == 'iceberg' %}
+            {{ adapter.iceberg_write(target_relation, sql, unique_key, partition_by, custom_location, strategy, table_properties) }}
+            {% set build_sql = "select * from glue_catalog." + target_relation.schema + "." + target_relation.identifier + " limit 1 "%}
         {% else %}
             {% set build_sql = create_table_as(False, target_relation, sql) %}
         {% endif %}
-      {% elif existing_relation_type == 'view' or full_refresh_mode %}
-        {{ drop_relation(target_relation) }}
+      {% elif existing_relation_type == 'view' or should_full_refresh() %}
+        {{ glue__drop_relation(target_relation) }}
```
Comment (nicor88) [inline dbt/include/glue/macros/materializations/incremental/incremental.sql:57]: nit: drop_relation should be enough. `glue__drop_relation` is automatically called when you call `drop_relation` as you are in the context of an adapter. See what we do here https://github.com/dbt-athena/dbt-athena/blob/main/dbt/include/athena/macros/materializations/models/table/table.sql#L66

Comment (sanga8) [reply dbt/include/glue/macros/materializations/incremental/incremental.sql]: thanks

**Review by menuetb — APPROVED** [review]
